api/lib/domain/services/algorithm-methods/flash.js

```diff
@@ -32,10 +32,15 @@ function getPossibleNextChallenges({
     forcedCompetences = [],
     challengesBetweenSameCompetence = 0,
     minimalSuccessRate = 0,
+    limitToOneQuestionPerTube = false,
   } = {},
 } = {}) {
   let nonAnsweredChallenges = getChallengesForNonAnsweredSkills({ allAnswers, challenges });
 
+  if (limitToOneQuestionPerTube) {
+    nonAnsweredChallenges = getChallengesForNonAnsweredTubes({ allAnswers, challenges });
+  }
```

Comment:
Pas besoin d'aller chercher 2 fois les challenges si on doit limite par tube

```suggestion
  let nonAnsweredChallenges = [];
    
  if (limitToOneQuestionPerTube) {
  nonAnsweredChallenges.push(...getChallengesForNonAnsweredSkills({ allAnswers, challenges }));
} else {
   nonAnsweredChallenges.push(...getChallengesForNonAnsweredTubes({ allAnswers, challenges }));
}

    
  }
```